src/openrct2/scenario/Scenario.cpp

```diff
@@ -427,10 +427,12 @@ static int32_t scenario_create_ducks()
     int32_t i, j, r, c, x, y, waterZ, centreWaterZ, x2, y2;
 
     // Originally, this function generated coordinates from 64 to 191.
-    // It now generates coordinates from 0 to 255, so smaller maps may also spawn ducks.
+    // The duck create function is called with 2 tiles of randomness to the target coordinate.
+    // The duck create function itself adds approximately 1 tile of randomness.
+    // To spawn ducks safely, the generated coordinates should thus be from 4 to 251.
     r = scenario_rand();
-    x = ((r >> 16) & 0xFFFF) % MAXIMUM_MAP_SIZE_TECHNICAL;
-    y = (r & 0xFFFF) % MAXIMUM_MAP_SIZE_TECHNICAL;
+    x = 4 + (((r >> 16) & 0xFFFF) % (MAXIMUM_MAP_SIZE_PRACTICAL - 6));
+    y = 4 + ((r & 0xFFFF) % (MAXIMUM_MAP_SIZE_PRACTICAL - 6));
```

Comment:
You can also use scenario_rand_max, just call it twice instead of using a single value.

Reply:
Since the game runs this function up to 100 times to try and create coordinates to spawn ducks at, it's probably less computationally intensive to only run the scenario_rand function once.

Reply:
Why do we bitmask the value twice? Seems unnecessary to me.

Reply:
I highly doubt scenario_rand_max will cause a bottleneck and its easier to read.